src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -3352,7 +3361,13 @@ bool DoorLockServer::HandleRemoteLockOperation(chip::app::CommandHandler * comma
 exit:
     if (!success && reason == OperationErrorEnum::kInvalidCredential)
     {
-        TrackWrongCodeEntry(endpoint);
+        HandleWrongCodeEntry(endpoint);
+    }
+
+    // Reset the wrong code retry attempts if a valid credential is presented
+    if (success && pinCode.HasValue())
+    {
+        ResetWrongCodeEntryAttempts(endpoint);
```

Comment:
So we do this unconditionally here, but only for unlock operations in SetLockState.  Why the inconsistency?

Also, worth documenting here that we need to do this because the relevant code in SetLockState will not be reached for a kRemote operation source.

Reply:
i should check for unlock here. Will add that

Reply:
should reset on both Lock/unlock